includes/class-wcpdf-main.php

```diff
@@ -913,6 +938,16 @@ public function set_phpmailer_validator( $mailArray ) {
 		return $mailArray;
 	}
 
+	/**
+	 * Logs the document creation to the order notes
+	 */
+	public function log_document_creation_to_order_notes( $document, $action ) {
+		if( ! empty( $document ) && ! empty( $action ) && ! empty( $order = $document->order ) ) {
+			$note = sprintf( __( 'PDF %s %s created on %s action.', 'woocommerce-pdf-invoices-packing-slips' ), $document->get_title() , $document->get_number()->get_plain(), $action );
```

Comment:
daisy-chaining is risky if you don't do a check on the return values, could lead to crashes